drain-service/drain_cp_inferencing.py

```diff
@@ -0,0 +1,269 @@
+        anomaly_df = df[df["anomaly_level"] == "Anomaly"]
+        if len(anomaly_df) == 0:
+            logging.info("No anomalies in this payload")
+        else:
+            try:
+                async for ok, result in async_streaming_bulk(
+                        es,
+                        doc_generator_anomaly(
+                            anomaly_df[["_id", "_op_type", "_index", "drain_control_plane_template_matched", "anomaly_level"]]
+                        ),
+                        max_retries=1,
+                        initial_backoff=1,
+                        request_timeout=5,
+                ):
+                    action, result = result.popitem()
+                    if not ok:
+                        logging.error("failed to {} document {}".format())
+                logging.info(f"Updated {len(anomaly_df)} anomalies in ES")
+            except (BulkIndexError, ConnectionTimeout, TimeoutError) as exception:
+                logging.error(
+                    "Failed to index data. Re-adding to logs_to_update_in_elasticsearch queue"
+                )
+                logging.error(exception)
+                await queue.put(anomaly_df)
+            except TransportError as exception:
+                logging.info(f"Error in async_streaming_bulk {exception}")
+                if exception.status_code == "N/A":
+                    logging.info("Elasticsearch connection error")
+                    es = await setup_es_connection()
+
+        try:
+            # update normal logs in ES
+            async for ok, result in async_streaming_bulk(
+                    es,
+                    doc_generator(
+                        df[
+                            [
+                                "_id",
+                                "_op_type",
+                                "_index",
+                                "drain_control_plane_template_matched"
+                            ]
+                        ]
+                    ),
+                    max_retries=1,
+                    initial_backoff=1,
+                    request_timeout=5,
+            ):
+                action, result = result.popitem()
+                if not ok:
+                    logging.error("failed to {} document {}".format())
+            logging.info(f"Updated {len(df)} logs in ES")
+        except (BulkIndexError, ConnectionTimeout) as exception:
+            logging.error("Failed to index data")
+            logging.error(exception)
+            await queue.put(df)
+        except TransportError as exception:
+            logging.info(f"Error in async_streaming_bulk {exception}")
+            if exception.status_code == "N/A":
+                logging.info("Elasticsearch connection error")
+                es = await setup_es_connection()
+
+async def init_nats():
+    # This function initialized the connection to Nats.
+    logging.info("connecting to nats")
+    await nw.connect()
+
+
+async def wait_for_index():
```

Comment:
why do we need it? the index should be ready before this pod starts?

Reply:
We should keep it since we want to make sure Elasticsearch is up and running before we try and index it.